numpy/core/src/umath/_umath_tests.c.src

```diff
@@ -603,19 +603,19 @@ UMath_Tests_test_dispatch(PyObject *NPY_UNUSED(dummy), PyObject *NPY_UNUSED(dumm
     NPY_CPU_DISPATCH_CALL_XB(highest_func_xb = _umath_tests_dispatch_func, ());
     NPY_CPU_DISPATCH_CALL_XB(highest_var_xb  = _umath_tests_dispatch_var);
 
-    PyObject *dict = PyDict_New(), *item;
+    PyObject *dict = PyDict_New(), *item = NULL;
     if (dict == NULL) {
         return NULL;
     }
     /**begin repeat
      * #str = func, var, func_xb, var_xb#
     */
-    item = PyUnicode_FromString(highest_@str@);
```

Comment:
This looks like the original was OK. Could move the declaration/assignment down here.

Reply:
The code is in a `/**begin repeat`loop. I missed that also first, but that means it gets repeated 3 times.  The (completely fine) alternative would be to put a Py_DECREF at the end of the loop.

Which, in hindsight is probably prettier...

Reply:
Pushed a commit to simplifiy this, no need for the big `SETREF` gun...